shared/src/persistence/elasticsearch/getDocketNumbersByStatusAndByJudge.ts

```diff
@@ -1,50 +1,64 @@
-import { search } from './searchClient';
+import {
+  CavAndSubmittedCaseResponseType,
+  JudgeActivityReportCavAndSubmittedCasesRequest,
+} from '../../../../web-client/src/presenter/judgeActivityReportState';
+import { MAX_ELASTICSEARCH_PAGINATION } from '../../business/entities/EntityConstants';
+import { QueryDslQueryContainer } from '@opensearch-project/opensearch/api/types';
+import { formatResults } from './searchClient';
 
-/**
- * getDocketNumbersByStatusAndByJudge
- * @param {object} providers the providers object containing applicationContext
- * @param {string} providers.applicationContext application context
- * @param {string} providers.judgeName judge
- * @param {string} providers.statuses case statuses
- * @returns {array} array of docketNumbers with field based on source
- */
 export const getDocketNumbersByStatusAndByJudge = async ({
   applicationContext,
-  judgeName,
-  statuses,
-}) => {
+  params,
+}: {
+  applicationContext: IApplicationContext;
+  params: JudgeActivityReportCavAndSubmittedCasesRequest;
+}): Promise<CavAndSubmittedCaseResponseType> => {
   const source = ['docketNumber'];
 
-  const searchParameters = {
+  const mustFilters: QueryDslQueryContainer[] = [];
+  const filters: QueryDslQueryContainer[] = [
+    {
+      terms: { 'status.S': params.statuses },
+    },
+  ];
+
+  if (params.judges.length) {
+    const shouldArray: Object[] = [];
+    params.judges.forEach(judge => {
+      const associatedJudgeFilters = {
+        match_phrase: {
+          'associatedJudge.S': judge,
+        },
+      };
+      shouldArray.push(associatedJudgeFilters);
+    });
+    const shouldObject: QueryDslQueryContainer = {
+      bool: {
+        should: shouldArray,
+      },
+    };
+    mustFilters.push(shouldObject);
+  }
```

Comment:
i'm confused why you're mixing and matching `must` and `should` here. plus, you're not including `minimum_should_match`, and so the `should` isn't required.

why not do 

```
bool: {
  should: [
    {
      match_phrase: {
        'associatedJudge.S': judge,
      },
    },
    ...
  ],
  minimum_should_match: 1
}
```

instead of 

```
bool: {
  must: [
    {
      bool: {
        should: [
          {
            'associatedJudge.S': judge,
          },
        },    
        ...
      ],
    }
  ],
  minimum_should_match: 1
}
```